### PR TITLE
update prefs to support gnome 40

### PR DESCRIPTION
**Pull request by kgarner7**
Hopefully addreses #672 

Gtk4 introduced breaking changes to how the preferences are created. This PR seeks to handle those changes while maintaining backwards compatibility
-  Gtk.HBox and Gtk.VBox are dead, replaced with Gtk.Box with orientation 
- `border_width` is dead, replace it with margins of 10 (this seems to function similarly)
- `add`, `pack_start`, `pack_end` calls for Box have been replaced with `append` and `prepend`: these calls are aliased where appropriate
- Because packing is gone, spacing is now controlled by `set_homogeneous`; It has bigger windows and boxes, but seems to be the closest to maintaining packing?
- `show_all` is unneeded in new versions
- `get_all_children` is gone, so I use effectively an iterator to build and sort the array

Additionally fixes complaints in `extension.js` from eslint.

Tested on fedora 34, arch linux (in English, for ordering).

I'm not sure if this is the best/cleanest way to multiplex, but hopefully it is at least a good start?

### CODE REVIEW

File: system-monitor@paradoxxx.zero.gmail.com/prefs.js [hunk 1]
```diff
@@ -66,12 +69,49 @@ function check_sensors(sensor_type) {
...
+
+    return aliasBox;
+}
 
 const ColorSelect = class SystemMonitor_ColorSelect {
     constructor(name) {
         this.label = new Gtk.Label({label: name + _(':')});
         this.picker = new Gtk.ColorButton();
-        this.actor = new Gtk.HBox({spacing: 5});
+        this.actor = box(5);
```
Comment (p91paul) [inline system-monitor@paradoxxx.zero.gmail.com/prefs.js:114]: this one should have been box(true, 5) probably!

Comment (kgarner7) [reply system-monitor@paradoxxx.zero.gmail.com/prefs.js]: Thanks for that catch!

**Comment by p91paul** [comment]
Tested this branch, it mostly works, however the preferences window doesn't fit anymore in my 1366x768 laptop screen.
Perhaps you exaggerated some spacing?

**Comment by p91paul** [comment]
My advice, merge this, release to e.g.o., then fix the small annoyances once you stopped the flood of complaining users.

**Comment by kgarner7** [comment]
The spacing issue was caused by the color selects being expanded to a larger size when `homogeneous` was set to true; this consequently scaled up the size of all other horizontal boxes. 

I've modified `box` to conditionally apply spacing, so it should no longer have that issue. Unfortunately this makes colors left-aligned instead of center aligned, but it should work on 1366x768 and 1280x (preserving packing for rows). It is too large for 1024x, which would require different spacing/packing options.

**Comment by p91paul** [comment]
Tested, looks good enough for me.
The window has always been pretty large, so probably 1024x users (if any) won't actually see a regression.

**Comment by kgarner7** [comment]
Sorry for delayed response. I think it should be stable

**Comment by Digitalone1** [comment]
If it's good, please merge.

**Comment by chrisspen** [comment]
I'm testing locally.

**Pull request merged**